src/tests/algorithms_test/pid_MergeParticleID.cc

```diff
@@ -0,0 +1,250 @@
+  for(int i=0; i<=pid.getNpe(); i++)
+    pid.addToThetaPhiPhotons(edm4hep::Vector2f{-M_PI/4, 0});
+
+  // collection 2 ------
+  auto coll_cherenkov_2 = std::make_unique<edm4eic::CherenkovParticleIDCollection>();
+
+  pid = coll_cherenkov_2->create();
+  pid.setChargedParticle(tracks->at(1)); // track 1 first
+  pid.setNpe(4);
+  pid.setRefractiveIndex(1.5);
+  pid.setPhotonEnergy(3e-9);
+  pid_hyp.PDG    = 211;
+  pid_hyp.npe    = 3;
+  pid_hyp.weight = 200;
+  pid.addToHypotheses(pid_hyp);
+  pid_hyp.PDG    = 321;
+  pid_hyp.npe    = 2;
+  pid_hyp.weight = 90;
+  pid.addToHypotheses(pid_hyp);
+  for(int i=0; i<=pid.getNpe(); i++)
+    pid.addToThetaPhiPhotons(edm4hep::Vector2f{M_PI/4, 0});
+
+  pid = coll_cherenkov_2->create();
+  pid.setChargedParticle(tracks->at(0));
+  pid.setNpe(6);
+  pid.setRefractiveIndex(1.3);
+  pid.setPhotonEnergy(4e-9);
+  pid_hyp.PDG    = 321;
+  pid_hyp.npe    = 4;
+  pid_hyp.weight = 70;
+  pid.addToHypotheses(pid_hyp);
+  pid_hyp.PDG    = 211;
+  pid_hyp.npe    = 1;
+  pid_hyp.weight = 80;
+  pid.addToHypotheses(pid_hyp);
+  pid_hyp.PDG    = 11; // an additional hypothesis
+  pid_hyp.npe    = 1;
+  pid_hyp.weight = 60;
+  pid.addToHypotheses(pid_hyp);
+  for(int i=0; i<=pid.getNpe(); i++)
+    pid.addToThetaPhiPhotons(edm4hep::Vector2f{-M_PI/4, 0});
+
+  // Cherenkov PID tests
+  //----------------------------------------------------------
+
+  std::vector<const edm4eic::CherenkovParticleIDCollection*> coll_cherenkov_list = {
+    coll_cherenkov_1.get(),
+    coll_cherenkov_2.get()
+  };
+
+  auto find_cherenkov_pid_for_track = [] (const auto& coll, auto trk) {
+    for(auto obj : *coll) {
+      if(obj.getChargedParticle().id() == trk.id())
+        return obj;
+    }
+    throw std::runtime_error("ERROR: cannot find CherenkovParticleID given track");
+  };
+
+  // additive weights
+  SECTION("merge CherenkovParticleID: add hypothesis weights") {
+
+    eicrecon::MergeParticleID algo;
+    eicrecon::MergeParticleIDConfig cfg;
+    cfg.mergeMode = eicrecon::MergeParticleIDConfig::kAddWeights;
+    algo.applyConfig(cfg);
+    algo.AlgorithmInit(logger);
+
+    auto result = algo.AlgorithmProcess(coll_cherenkov_list);
+    auto pid_0  = find_cherenkov_pid_for_track(result, tracks->at(0));
+    auto pid_1  = find_cherenkov_pid_for_track(result, tracks->at(1));
+
+    REQUIRE_THAT( pid_0.getNpe(), Catch::Matchers::WithinAbs(10+6, EPSILON) );
+    REQUIRE_THAT( pid_1.getNpe(), Catch::Matchers::WithinAbs(11+4, EPSILON) );
+
+    REQUIRE_THAT( pid_0.getRefractiveIndex(), Catch::Matchers::WithinAbs((10*1.05+6*1.3)/(10+6), EPSILON) );
+    REQUIRE_THAT( pid_1.getRefractiveIndex(), Catch::Matchers::WithinAbs((11*1.06+4*1.5)/(11+4), EPSILON) );
+
+    REQUIRE_THAT( pid_0.getPhotonEnergy(), Catch::Matchers::WithinAbs((10*3e-9+6*4e-9)/(10+6), EPSILON) );
+    REQUIRE_THAT( pid_1.getPhotonEnergy(), Catch::Matchers::WithinAbs((11*4e-9+4*3e-9)/(11+4), EPSILON) );
+
+    for(auto hyp : pid_0.getHypotheses()) {
```

Comment:
Can we also check the number of produced hypotheses?

Reply:
[71bdeac](https://github.com/eic/EICrecon/pull/730/commits/71bdeac5242a44df2450fd2e872edc03ac7713b3)

Reply:
letting CI check the syntax...